### PR TITLE
Switched create channel to new description field

### LINKED ISSUE #3715: Switch to open-discussions description field for channel creation
We currently set `public_description`, we should instead use `description` since that's what we display on open-discussions.

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Closes #3715 

#### What's this PR do?
Switches to the new field we're using for channel descriptions

#### How should this be manually tested?
Create a new channel using markdown in description, ensure this shows up in the channel sidebar once created.

### CODE REVIEW

File: static/js/flow/discussionTypes.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ type ChannelType = "private" | "public";
 export type ChannelInputs = {
   name:               string,
   title:              string,
-  public_description: string,
+  description: string,
```
Comment (noisecapella) [inline static/js/flow/discussionTypes.js]: Can you adjust the indentation here and below? I guess the reformatter isn't aware of flow type declarations

Comment (rhysyngsun) [reply static/js/flow/discussionTypes.js]: Yeah, these need to be manually formatted actually cause the formatter actually removes the alignments.

Comment (noisecapella) [reply static/js/flow/discussionTypes.js]: That's fine, let's just leave it then

Comment (rhysyngsun) [reply static/js/flow/discussionTypes.js]: Fixed it anyway since it was small/quick

**Pull request merged**